hw/ip/pwrmgr/doc/_index.md

```diff
@@ -0,0 +1,233 @@
+There is thus no need to sequence its operation like the others.
+
+The power controller observes reset requests only in two states - the slow FSM LOW\_POWER state and the fast FSM ACTIVE state - these are considered static states of the system.
+When a reset request is received during slow FSM LOW\_POWER state, the system begins its usual power up sequence even if a wakeup has not been received.
+
+When a reset request is received during fast FSM ACTIVE state, either because the system was already active or exited low power due to the reset request,  the fast FSM asserts resets immediately and transitions back through its initialization process.
+Note in this case, the slow FSM is “not activated” and remains in its IDLE state.
+
+
+### Reset Requests Received During Other States
+
+All other states in the slow / fast FSM are considered transition states.
+Resets are not observed in other states because the system will always be transitioning towards one of the steady states (the system is in the process of powering down or powering up).
+Once a steady state is reached, the reset request is then observed and action taken.
+
+### Reset Recording
+
+Of the resets directly handled by the power manager (peripheral reset request, escalation and low power entry), the power manager sequences only 1 at a time (see state diagrams).
+This means if all reset requests happen at approximately the same time, the power manager will handle in a first come first serve manner.
+When the handling of the first is completed, it observes there is another pending request and handles accordingly.
+This is done because low power resets and peripheral requested resets lead to different behaviors.
+When the power manager commits to handling a specific request, it sends a reset hint to the reset manager to inform it of what reasons to record should the processor reset.
+
+If for example a low power entry request is seen first by the power manager, it transitions the system to a low power state and informs the reset manager to record “low power exit” as the reset reason.
+If during this transition a watchdog request is observed, the power manager then uses the watchdog as a wakeup cause to resume the system and handle the associated request.
+As this request is handled, the reset manager records a second reset reason of “watchdog”.
+
+Ultimately when control is returned to software, it sees two reset reasons and must handle accordingly.
+
+
+## Wakeup Recording
+
+Similar to [reset handling](#Reset-Request-Handling), wakeup signals are only observed during slow FSM low power; however its recording is continuous until explicitly disabled by software.
+
+The wakeup recording begins when the fast FSM transitions out of ACTIVE state and continues until explicitly disabled by software.
+This ensures wakeup recording is continuous until software has set up the appropriate peripherals to handle future events.
```

Comment:
This last sentence feels a little circular ("the recording is continuous to ensure that it's continuous"). Presumably the point is to avoid missing anything, so maybe "This ensures that no wakeup events are missed by coming in before software has set up ..." ?

Reply:
fixed